### PR TITLE
MCR-3176 fix pmd issues in mycore-iview2

### PR DESCRIPTION
**Pull request by yagee-de**
[Link to jira](https://mycore.atlassian.net/browse/MCR-3176).

Should replace https://github.com/MyCoRe-Org/mycore/pull/2245

### CODE REVIEW

File: mycore-iview2/src/main/java/org/mycore/iview2/iiif/MCRIVIEWIIIFImageImpl.java [hunk 1]
```diff
@@ -161,58 +152,90 @@ public BufferedImage provide(String identifier,
...
+            (int) Math.round(region.y2() * zoomLevelScale));
 
         // now we detect the tiles to draw!
-        int x1Tile = (int) Math.floor(x1 / 256),
-            y1Tile = (int) Math.floor(y1 / 256),
-            x2Tile = (int) Math.ceil(x2 / 256),
-            y2Tile = (int) Math.ceil(y2 / 256);
-
+        double tileSizeFactor = zoomLevelScale / 256;
+        MCRIIIFImageSourceRegion sourceTiles = new MCRIIIFImageSourceRegion(
```
Comment (Antoniafriedrich) [inline mycore-iview2/src/main/java/org/mycore/iview2/iiif/MCRIVIEWIIIFImageImpl.java:130]: Why was the formula for the sourceTiles changed?

File: mycore-iview2/src/main/java/org/mycore/iview2/iiif/MCRIVIEWIIIFImageImpl.java [hunk 2]
```diff
@@ -161,58 +152,90 @@ public BufferedImage provide(String identifier,
...
+        double rotatationRadians, int sourceZoomLevel, double zoomLevelScale, MCRIIIFImageSourceRegion scaledRegion) {
+        if (rotation.mirrored()) {
+            graphics.scale(-1, 1);
+            graphics.translate(-rotatedSize.width(), 0);
         }
 
-        return targetImage;
+        graphics.translate((rotatedSize.width() - targetSize.width()) / 2,
+            (rotatedSize.height() - targetSize.height()) / 2);
+        int xt = (int) ((targetSize.width() - 1) / 2);
```
Comment (Antoniafriedrich) [inline mycore-iview2/src/main/java/org/mycore/iview2/iiif/MCRIVIEWIIIFImageImpl.java]: Casting xt and yt to 'int' is redundant.

File: mycore-iview2/src/main/java/org/mycore/iview2/frontend/MCRIView2Commands.java [hunk 3]
```diff
@@ -415,8 +424,7 @@ private static void deleteFileAndEmptyDirectories(Path file) throws IOException
         }
         if (Files.isDirectory(file)) {
             try (DirectoryStream<Path> directoryStream = Files.newDirectoryStream(file)) {
```
Comment (Mewel) [inline mycore-iview2/src/main/java/org/mycore/iview2/frontend/MCRIView2Commands.java:426]: boolean isEmptyDirectory = Files.list(file).findAny().isEmpty()
is much shorter than using a directory stream

**Review by Mewel — CHANGES_REQUESTED** [review]

File: mycore-iview2/src/main/java/org/mycore/iview2/iiif/MCRIVIEWIIIFImageImpl.java (hunk 2 above)
Comment (yagee-de) [reply mycore-iview2/src/main/java/org/mycore/iview2/iiif/MCRIVIEWIIIFImageImpl.java]: Yep, good point the final type requested is also a double, so this makes no sense at all.

File: mycore-iview2/src/main/java/org/mycore/iview2/frontend/MCRIView2Commands.java (hunk 3 above)
Comment (yagee-de) [reply mycore-iview2/src/main/java/org/mycore/iview2/frontend/MCRIView2Commands.java:426]: Thanks, implemented in https://github.com/MyCoRe-Org/mycore/pull/2277/commits/3f0661ad939505ff717d05573fbf415811fe67e5

**Review by Mewel — APPROVED** [review]

**Pull request merged**